sklearn/feature_extraction/tests/test_text.py

```diff
@@ -351,6 +351,30 @@ def test_fit_countvectorizer_twice():
     assert X1.shape[1] != X2.shape[1]
 
 
+def test_countvectorizer_custom_token_pattern():
+    corpus = [
+        'This is the 1st document in my corpus.',
+        'This document is the 2nd sample.',
+        'And this is the 3rd one.',
+        'Is this the 4th document?',
+    ]
+    # With 1 capturing group, capturing the word after an ordinal
+    token_pattern = r"[0-9]{1,3}(?:st|nd|rd|th)\s\b(\w{2,})\b"
+    vectorizer = CountVectorizer(token_pattern=token_pattern)
+    vectorizer.fit_transform(corpus)
+    expected = ['document', 'one', 'sample']
+    assert vectorizer.get_feature_names() == expected
+    # With 2 capturing groups, capturing the ordinal and the word after it
+    token_pattern = r"([0-9]{1,3}(?:st|nd|rd|th))\s\b(\w{2,})\b"
+    message = "more than 1 capturing group in token pattern"
+
+    def func(token_pattern):
+        vectorizer = CountVectorizer(token_pattern=token_pattern)
+        vectorizer.fit_transform(corpus)
+
+    assert_raise_message(ValueError, message, func, token_pattern)
```

Comment:
We usually try to test a more minimal context that will error. That is, rather than func here, we would have vectorizer.fit

Reply:
Done!